File: src/types/card.ts
export interface Deck {
  id: number
  name: string
  level: string
  description: string | null
  total_cards: number
  is_active: number
  created_at: string
}

export interface Card {
  id: number
  deck_id: number
  front: string
  back: string
  example_sentence: string | null
  phonetic: string | null
  difficulty: number
  stability: number
  retrievability: number
  last_review_at: string | null
  next_review_at: string | null
  review_count: number
  lapses: number
  state: 'new' | 'learning' | 'review' | 'relearning'
  created_at: string
}

File: src-tauri/src/models/card.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Deck {
    pub id: i64,
    pub name: String,
    pub level: String,
    pub description: Option<String>,
    pub total_cards: i64,
    pub is_active: i64,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Card {
    pub id: i64,
    pub deck_id: i64,
    pub front: String,
    pub back: String,
    pub example_sentence: Option<String>,
    pub phonetic: Option<String>,
    pub difficulty: f64,
    pub stability: f64,
    pub retrievability: f64,
    pub last_review_at: Option<String>,
    pub next_review_at: Option<String>,
    pub review_count: i64,
    pub lapses: i64,
    pub state: String,
    pub created_at: String,
}
